driver/src/transport.rs

```diff
@@ -67,10 +67,10 @@ impl HttpTransportInner {
         let request = serde_json::to_string(&request)?;
         debug!("[id:{}] sending request: '{}'", id, &request,);
 
-        let (response, content) = self
-            .post_json(request)
-            .await
-            .map_err(|err| Web3Error::Transport(err.to_string()))?;
+        let (response, content) = self.post_json(request).await.map_err(|err| {
+            warn!("[id:{}] returned an error: '{}'", id, err.to_string());
+            Web3Error::Transport(err.to_string())
+        })?;
```

Comment:
Would it be better to instead (or in addition) of warning make the whole warning string the new error message?

Reply:
Given that the ID is something internal to the request, I'm not sure there is much value in leaking it to the higher layer. As in, if you don't have logging for the transport module enabled you don't see which requests maps to which id and thus the information that request 42 error-ed doesn't give you any value.